src/coreComponents/mesh/mpiCommunications/mpiCommunicationUtils.hpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CommID.hpp
+ */
+
+#ifndef GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+#define GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+
+#include "common/DataTypes.hpp"
+
+namespace geosx
+{
+
+struct FieldIdentifiers
+{
+  void addFields( FieldLocation const location, std::vector<string> const fieldNames )
+  {
+    string const key = generateKey(location);
+    addFields(fieldNames, key);
+  }
+
+  void addElementFields( std::vector<string> const fieldNames, std::vector<string> const regionNames )
+  {
+    for (string const & regionName : regionNames)
+    {
+      string const key = generateKey( regionName );
+      addFields(fieldNames, key);
+    }
+  }
+
+  std::map< string, array1d<string> > const & getFields() const
```

Comment:
@TotoGaz The changes are motivated by a longer term goal to treat elements and other objects in a similar manner. I view the most straight forward way to specify a field, or a group of fields is to have a `map<string,vector<string>>` where the key is the path to the object that contains the fields, and the value are the names of the fields on that object. Rather than having a separate container that may or may not be empty  to hold region names, this simply embeds the information into the key.

Reply:
If we want to go that way, it appears useful to have a dedicated strong `Key` class that would hold the appropriate information in a convenient way, could return the `std::optional` when we're not sure we have something, could store the information in `union` for mutually exclusive fields, etc.

Plain `string` is flat, colorless, cannot be encapsulated.
For end users to define it in input files, it's surely OK, but when writing C++ code for devs, it's less obvious.